lib/spack/spack/build_environment.py

```diff
@@ -141,17 +142,21 @@ def clean_environment():
     # can affect how some packages find libraries.  We want to make
     # sure that builds never pull in unintended external dependencies.
     env.unset('LD_LIBRARY_PATH')
-    env.unset('CRAY_LD_LIBRARY_PATH')
     env.unset('LIBRARY_PATH')
     env.unset('CPATH')
     env.unset('LD_RUN_PATH')
     env.unset('DYLD_LIBRARY_PATH')
     env.unset('DYLD_FALLBACK_LIBRARY_PATH')
 
-    # Remove all pkgconfig stuff from craype
-    for varname in os.environ.keys():
-        if 'PKGCONF' in varname:
-            env.unset(varname)
+    # Test the host architecture because old cray systems (i.e. Blue Waters)
```

Comment:
IMO this comment could be clearer, for example:

> On Cray systems newer than CNL5, unset CRAY_LD_LIBRARY_PATH to avoid interference with Spack dependencies. CNL5 (e.g. Blue Waters) requires these variables to be set.

My main complaint with the comment is that it mentions the exception before the rule, so if you don't think my rewording is good, I think any rewording will be good as long as it mentions the rule up front (and also I think this comment assumes that CNL5 is the oldest Cray version that will be encountered).